babynames.py

```diff
@@ -42,29 +45,66 @@
 def extract_names(filename):
     """
     Given a single file name for babyXXXX.html, returns a single list starting
-    with the year string followed by the name-rank strings in alphabetical order.
+    with the year string followed by the name-rank strings in alphabetical
+    order.
     ['2006', 'Aaliyah 91', Aaron 57', 'Abagail 895', ' ...]
     """
     names = []
-    # +++your code here+++
-    return names
+
+    names.append(re.findall("\d\d\d\d", filename)[0])
```

Comment:
I totally thought of using the filename to get the date!  Ended up going with parsing through the code because it was the suggested regex expression.  Your version is much cleaner-looking and easier to read :)